notebooks/4_visualize_forecast.ipynb

```diff
@@ -0,0 +1,1082 @@
+   "outputs": [],
+   "source": [
+    "destinations_all = list(set(first_intros_all[\"Destination\"]))\n",
+    "min_intro_prop = 0.5\n",
+    "destinations = []\n",
+    "for i in range(len(destinations_all)):\n",
+    "    if (\n",
+    "        len(\n",
+    "            first_intros_all.loc[first_intros_all[\"Destination\"] == destinations_all[i]]\n",
+    "        )\n",
+    "        > num_runs * min_intro_prop\n",
+    "    ):\n",
+    "        destinations.append(destinations_all[i])\n",
+    "num_destinations = len(destinations)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Count origins for each timestep\n",
+    "origin_countries_by_ts = (\n",
+    "    pd.DataFrame(index=countries.iloc[:, 0], columns=sim_years)\n",
+    ").fillna(0)\n",
+    "origins = (org_dest_all.groupby([\"Origin\", \"TS\"]).count()).reset_index().fillna(0)\n",
+    "bridgehead_years = range(origins.TS.min(), origins.TS.max()+1)\n",
+    "\n",
+    "for i in range(len(origins)):\n",
+    "    origin = origins.iloc[i, :]\n",
+    "    origin_countries_by_ts.loc[origin.Origin, origin.TS] = origin.Destination"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create heatmap of bridgehead introductions\n",
+    "origin_countries_by_ts_filtered = origin_countries_by_ts[\n",
+    "    origin_countries_by_ts.sum(axis=1) > 0][bridgehead_years]\n",
+    "fig, ax = plt.subplots(figsize=(12, 8))\n",
+    "plt.subplots_adjust(left=0.22, right=1, top=0.92)\n",
+    "res = sns.heatmap(\n",
+    "    origin_countries_by_ts_filtered.drop(native_countries_list, errors=\"ignore\"),\n",
+    "    cmap=sns.color_palette(\"light:#31688e\", as_cmap=True),\n",
+    "    linewidths=0.30,\n",
+    "    annot=False,\n",
+    "    cbar_kws={\"label\": f\"Total Outgoing Transmissions Over {num_runs} Model Runs\"},\n",
+    ")\n",
+    "res.set_xticklabels(res.get_xmajorticklabels(), fontsize=14)\n",
+    "res.set_yticklabels(res.get_ymajorticklabels(), fontsize=14)\n",
+    "ax.figure.axes[-1].yaxis.label.set_size(14)\n",
+    "plt.title(\"Exports from Bridgehead Populations\", fontsize=20, pad=15)\n",
+    "plt.ylabel(\"\")\n",
+    "plt.xlabel(\"Year\", fontsize=14)\n",
+    "\n",
+    "plt.savefig(f\"{results_dir}/figs/bridgehead_sources.png\", dpi=600)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Histograms of introductions by year, for individual countries:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plot histograms of first intros by destination\n",
+    "fig, axs = plt.subplots(\n",
+    "    2, math.ceil(num_destinations / 2), sharey=True, sharex=True, figsize=(12, 6)\n",
+    ")\n",
+    "fig.subplots_adjust(hspace=0.35, wspace=0.15, top=0.82)\n",
```

Comment:
Add lines to hide last subplot if the number of destinations is odd:

if num_destinations % 2 != 0:
axs[-1, -1].axis('off')